packages/node/src/handlers.ts

```diff
@@ -391,11 +392,55 @@ export function requestHandler(
     const local = domain.create();
     local.add(req);
     local.add(res);
-    local.on('error', next);
+    local.on('error', err => {
+      // Request mode session capturing functionality is added here so that we are able to catch errors even if Sentry's
+      // error handler is not used
+      if (isAutosessionTrackingEnabled()) {
+        const scope = getCurrentHub().getStackTop().scope;
+        if (scope) {
+          /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+          const requestSessionStatus = (scope as any)._requestSessionStatus;
+          if (requestSessionStatus !== undefined) {
+            (scope as any)._requestSessionStatus = RequestSessionStatus.Errored;
+          }
+          /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+        }
+      }
+      next(err);
+    });
+
     local.run(() => {
-      getCurrentHub().configureScope(scope =>
-        scope.addEventProcessor((event: Event) => parseRequest(event, req, options)),
-      );
+      const currentHub = getCurrentHub();
+      currentHub.configureScope(scope => {
+        scope.addEventProcessor((event: Event) => parseRequest(event, req, options));
+        if (isAutosessionTrackingEnabled()) {
+          const scope = currentHub.getScope();
+          if (scope) {
+            /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+            // Set `_requestSessionStatus` to Ok, at the beginning of the request and it is required to place it
+            // on the response as well because for some reason the scope is cleared in the `errorHandler`
+            // ToDo: Investigate why the scope does not propagate to the errorHandler
+            (scope as any)._requestSessionStatus = RequestSessionStatus.Ok;
+            (res as any).__sentry_requestSessionStatus = RequestSessionStatus.Ok;
+
+            /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+          }
+        }
+      });
+
+      res.once('finish', () => {
+        // Calling captureRequestSession to capture request mode session at the end of the request
+        setImmediate(() => {
+          if (isAutosessionTrackingEnabled()) {
```

Comment:
Shouldn't we check if session tracking is enabled before calling `setImmediate`?